applications/StructuralMechanicsApplication/custom_elements/small_displacement_bbar.cpp

```diff
@@ -0,0 +1,953 @@
+            TotalArea += IntegrationWeight;
+
+            for (unsigned int i = 0; i < number_of_nodes * 2; i++) {
+                // Bh = Bh + sum(Bs(1:3, : )) * wg(iPG) * detJ;
+                rVariables.Bh(i) += (rVariables.B(0, i) + rVariables.B(1, i)) * IntegrationWeight;
+            }
+        }
+        for (unsigned int i = 0; i < number_of_nodes * 2; i++) {
+            rVariables.Bh(i) /= TotalArea;
+        }
+    }
+    else {
+        double TotalVolume = 0.0;
+        for (unsigned int PointNumber = 0; PointNumber < integration_points.size(); PointNumber++) {
+            // Compute element kinematics B, F, DN_DX ...
+            CalculateKinematicVariables(rVariables, PointNumber, integration_points);
+            double IntegrationWeight = rVariables.detJ0 * integration_points[PointNumber].Weight();
+            TotalVolume += IntegrationWeight;
+
+            for (unsigned int i = 0; i < number_of_nodes; i++) {
+                // Bh = Bh + sum(Bs(1:3, : )) * wg(iPG) * detJ;
+                rVariables.Bh(i * 3 + 0) += rVariables.B(0, i * 3 + 0) * IntegrationWeight;
+                rVariables.Bh(i * 3 + 1) += rVariables.B(1, i * 3 + 1) * IntegrationWeight;
+                rVariables.Bh(i * 3 + 2) += rVariables.B(2, i * 3 + 2) * IntegrationWeight;
+            }
+        }
+        for (unsigned int i = 0; i < number_of_nodes * 3; i++) {
+            rVariables.Bh(i) /= TotalVolume;
+        }
+    }
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Matrix SmallDisplacementBbar::ComputeEquivalentF(const Vector& rStrainTensor)
+{
+    const unsigned int dim = GetGeometry().WorkingSpaceDimension();
+    Matrix F(dim,dim);
+
+    if(dim == 2) {
+        F(0,0) = 1.0 + rStrainTensor(0);
+        F(0,1) = 0.5 * rStrainTensor(2);
+        F(1,0) = 0.5 * rStrainTensor(2);
+        F(1,1) = 1.0 + rStrainTensor(1);
+    } else {
+        F(0,0) = 1.0 + rStrainTensor(0);
+        F(0,1) = 0.5 * rStrainTensor(3);
+        F(0,2) = 0.5 * rStrainTensor(5);
+        F(1,0) = 0.5 * rStrainTensor(3);
+        F(1,1) = 1.0 + rStrainTensor(1);
+        F(1,2) = 0.5 * rStrainTensor(4);
+        F(2,0) = 0.5 * rStrainTensor(5);
+        F(2,1) = 0.5 * rStrainTensor(4);
+        F(2,2) = 1.0 + rStrainTensor(2);
+    }
+
+    return F;
+}
+
+//************************************************************************************
+//************************************************************************************
+
+void SmallDisplacementBbar::CalculateOnIntegrationPoints(
+        const Variable<double>& rVariable,
+        std::vector<double>& rOutput,
+        const ProcessInfo& rCurrentProcessInfo)
+{
+    if (rOutput.size() != GetGeometry().IntegrationPoints(  ).size())
+        rOutput.resize(GetGeometry().IntegrationPoints(  ).size());
+
```

Comment:
If the variables output are already defined in the base class just add an else at the end with 
else {
BaseType::CalculateOnIntegrationPoints(rVariable,rOutput, rCurrentProcessInfo);
}

Reply:
Sorry, you already did, forgot that